generators/Exercises/ExerciseCollection.cs

```diff
@@ -35,13 +36,14 @@ private IEnumerable<Exercise> GetExercises()
             foreach (var exercise in TrackConfigFile.GetExercises())
             {
                 var exerciseName = exercise.Slug.ToExerciseName();
+
                 if (exercise.Deprecated)
                     yield return new DeprecatedExercise(exerciseName);
                 else if (HasNoCanonicalData(exerciseName))
                     yield return new MissingDataExercise(exerciseName);
                 else if (IsNotImplemented(exerciseName))
                     yield return new UnimplementedExercise(exerciseName);
-                else if (IsOutdated(exerciseName))
+                else if (IsOutdated(exerciseName) && !_options.ShouldGenerate)
```

Comment:
If you replace `!_options.ShouldGenerate` with ` && _options.Status == GeneratorStatus.Outdated`, it will ensure that outdated exercises are only returned when the outdated status is requested. The current code works when specifying a single exercise, but it does not work when just doing `dotnet run` to regenerate all exercises. In that case, the outdated exercises are listed as outdated and are not updated.

Another possible approach we could consider is adding a `Outdated` property to the `GeneratorExercise` class, and then add some additional logic in the `RegenerateTestClass` method to handle this case. I think I would actually prefer this latter approach, as an exercise can semantically be both a generator exercise _and_ an outdated exercise. What do you think?

Reply:
I've merged a [temporary fix](#686) by the way, so take your time to implement this.